### PR TITLE
allow configuring hcm settings

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

File: projects/gloo/pkg/translator/utils.go [hunk 1]
```diff
@@ -1,7 +1,51 @@
...
+
+	return s, nil
+}
+
+func ParseConfig(c configObject, config proto.Message) error {
+	any := c.GetTypedConfig()
+	if any != nil {
+		return types.UnmarshalAny(any, config)
+	}
+	structt := c.GetConfig()
```
Comment (mitchdraft) [inline projects/gloo/pkg/translator/utils.go]:
```suggestion
	struct := c.GetConfig()
```

File: projects/gloo/pkg/translator/utils.go [hunk 2]
```diff
@@ -1,7 +1,51 @@
...
+	if structt != nil {
+		return envoyutil.StructToMessage(structt, config)
```
Comment (mitchdraft) [inline projects/gloo/pkg/translator/utils.go]:
```suggestion
		return envoyutil.StructToMessage(struct, config)
```

File: projects/gloo/pkg/plugins/hcm/plugin.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+			if filter.Name == envoyutil.HTTPConnectionManager {
+				// get config
+				var cfg envoyhttp.HttpConnectionManager
+				err := translatorutil.ParseConfig(&filter, &cfg)
+				// this should never error
+				if err != nil {
+					return err
+				}
+
+				cfg.UseRemoteAddress = hcmSettings.UseRemoteAddress
```
Comment (mitchdraft) [inline projects/gloo/pkg/plugins/hcm/plugin.go]: this is nested pretty deep, could break it into a function

File: projects/gloo/api/v1/plugins/hcm/hcm.proto [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+option go_package = "github.com/solo-io/gloo/projects/gloo/pkg/api/v1/plugins/hcm";
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/duration.proto";
+
+option (gogoproto.equal_all) = true;
+
+message HttpConnectionManagerSettings {
+    bool skip_xff_append = 1;
```
Comment (mitchdraft) [inline projects/gloo/api/v1/plugins/hcm/hcm.proto]: I feel like we should add comments to these for the generated docs

File: projects/gloo/pkg/translator/utils.go (hunk 2 above)
Comment (yuval-k) [reply projects/gloo/pkg/translator/utils.go]: struct is a key word; this is my equivalent of clazz in java

File: projects/gloo/pkg/plugins/hcm/plugin.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func copySeetings(cfg *envoyhttp.HttpConnectionManager, hcmSettings *hcm.HttpConnectionManagerSettings) {
```
Comment (mitchdraft) [inline projects/gloo/pkg/plugins/hcm/plugin.go]:
```suggestion
func copySettings(cfg *envoyhttp.HttpConnectionManager, hcmSettings *hcm.HttpConnectionManagerSettings) {
```

**Comment by benishak** [comment]
@yuval-k 
Thanks
i'm trying to configure HCM, I added this to Settings

```
kind: Settings
metadata:
  name: default
  namespace: gloo-system
spec:
  bindAddr: 0.0.0.0:9977
  httpConnectionManagerSettings:
    useRemoteAddress: true
```

I got error
```
reading crd spec into v1Settings: unknown field \"httpConnectionManagerSettings
```

**Pull request merged**